ext/drmgr/drmgr.h

```diff
@@ -165,6 +165,19 @@ typedef dr_emit_flags_t (*drmgr_insertion_cb_t)(void *drcontext, void *tag,
                                                 bool for_trace, bool translating,
                                                 void *user_data);
 
+/**
+ * Callback function for opcode based instrumentation. In particular, this callback
+ * is triggered only for specific instruction opcodes. This is done during the
+ * third stage, i.e., instrumentation insertion.
+ *
+ * See #dr_emit_flags_t for an explanation of the return value.  If
+ * any instrumentation pass requests DR_EMIT_STORE_TRANSLATIONS, they
+ * will be stored.
+ */
+typedef dr_emit_flags_t (*drmgr_opcode_insertion_cb_t)(void *drcontext, void *tag,
+                                                       instrlist_t *bb, instr_t *inst,
+                                                       bool for_trace, bool translating);
```

Comment:
How about including a `void *user_data`?  I wish we had done so for all event cb's up front (instead of having to add to some later as extended versions).

Reply:
Unlike cbs for instr insert instrumentation, opcode cbs are not paired with any prior analysis cbs for now. Shall I add the parameter and pass NULL?

Reply:
I meant a static param passed at registration time (yeah that is a little confusing vs the per-invocation user_data used in the combined instru cbs).

Reply:
Done.